src/main/java/org/folio/innreach/domain/service/impl/MARCTransformationOptionsSettingsServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package org.folio.innreach.domain.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.folio.innreach.domain.entity.FieldConfiguration;
+import org.folio.innreach.domain.entity.MARCTransformationOptionsSettings;
+import org.folio.innreach.domain.exception.EntityNotFoundException;
+import org.folio.innreach.domain.service.MARCTransformationOptionsSettingsService;
+import org.folio.innreach.dto.MARCTransformationOptionsSettingsDTO;
+import org.folio.innreach.mapper.MARCTransformationOptionsSettingsMapper;
+import org.folio.innreach.repository.MARCTransformationOptionsSettingsRepository;
+import org.springframework.data.domain.Example;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityExistsException;
+import java.util.ArrayList;
+import java.util.UUID;
+
+import static org.folio.innreach.domain.service.impl.ServiceUtils.centralServerRef;
+
+@RequiredArgsConstructor
+@Service
+public class MARCTransformationOptionsSettingsServiceImpl implements MARCTransformationOptionsSettingsService {
+
+  private final MARCTransformationOptionsSettingsRepository repository;
+  private final MARCTransformationOptionsSettingsMapper mapper;
+
+  private static final String TEXT_MARC_TRANSFORM_OPT_SET_WITH_ID = "MARC Transformation Options Settings with id: ";
+
+  @Override
+  public MARCTransformationOptionsSettingsDTO getMARCTransformOptSet(UUID centralServerId) {
+    var marcTransformOptSet = findMARCTransformOptSet(centralServerId);
+    return mapper.toMARCTransformationOptSetDto(marcTransformOptSet);
+  }
+
+  @Override
+  public MARCTransformationOptionsSettingsDTO createMARCTransformOptSet(UUID centralServerId, MARCTransformationOptionsSettingsDTO marcTransformOptSetDTO) {
+    repository.findById(centralServerId).ifPresent(marcTransformOptSet -> {
```

Comment:
this won't work as MARC settings has it's own generated id which differs from centralServerId. Please consider refactoring `findMARCTransformOptSet` to return `Optional<MARCTransformationOptionsSettings>` and move exception throwing out of that method. This way you can reuse it where its needed to find entity by server id.